core/src/main/java/org/apache/iceberg/avro/TypeToSchema.java

```diff
@@ -214,4 +216,58 @@ public Schema primitive(Type.PrimitiveType primitive) {
 
     return primitiveSchema;
   }
+
+  private static String sanitize(String name) {
+    if (validAvroName(name)) {
+      return name;
+    }
+    int length = name.length();
+    StringBuilder sb = new StringBuilder(name.length());
+    char first = name.charAt(0);
+    if (!(Character.isLetter(first) || first == '_')) {
+      sb.append(sanitizedValue(first));
+    } else {
+      sb.append(first);
+    }
+    for (int i = 1; i < length; i++) {
+      char character = name.charAt(i);
+      if (!(Character.isLetterOrDigit(character) || character == '_')) {
+        sb.append(sanitizedValue(character));
+      } else {
+        sb.append(character);
+      }
+    }
+    return sb.toString();
+  }
+
+  private static final Map<Character, String> SANITIZE_MAP = new ImmutableMap
+      .Builder<Character, String>()
+      .put('.', "__DOT__").put('#', "__HASH__")
```

Comment:
Because of this change 
`icebergSchema` will not always be equal to `toIcebergSchema(toAvro(icebergSchema))` when special chars exists in Iceberg schema name.

Do you think we need to reverse the mapping when converting Avro to IcebergSchema e.g `__HASH__` converted to `#` etc. so the original names in the Iceberg schema are preserved?

Reply:
I don't think that will be necessary, but it would probably be a good idea to add a property with the original Iceberg name, just in case we ever want to. That's easier than converting back because the name could actually be `__HASH__`.